src/app/_components/userProfile/UserProfile.tsx

```diff
@@ -1,10 +1,11 @@
-"use client";
-
-import { Session } from "next-auth";
-import { api } from "~/trpc/react";
+import QRCode from "react-qr-code"
+import { type Session } from "next-auth";
+import { type Team } from "@prisma/client";
+import SignOut from "~/app/_components/userProfile/signOut";
 
 interface UserProfileProps {
   session: Session;
+  team: Team;
 }
 
 export function UserProfile(props: UserProfileProps) {
```

Comment:
Prefer de-structured inline props, makes reading code easier and faster to debug. Consider interfaces only when there will be potential heavy re-use and/or composition.